blog/_posts/2023-12-08-migration-to-scala-3.md

```diff
@@ -0,0 +1,153 @@
+
+> Note that you can find further help from the community on [Scala Discord](https://discord.com/invite/scala).
+
+We don’t have to go into these details right now. Instead, let’s talk about the bigger picture and a few other things to remember.
+
+### Dealing with internal libraries
+
+There are multiple ways of sharing a library between Scala 2 and Scala 3 services.
+
+**The good news** is that you can use Scala 2.13 libraries from Scala 3 app and Scala 3 libraries from Scala 2.13 app.
+
+**The bad news** is that you can’t do it directly if you have shared transitive dependencies.
+
+**The good news** is that it’s pretty easy to deal with.
+
+Imagine you have an internal library (e.g., auth-handling, specific DB-connector, peculiar API) used by your existing Scala 2 services that will also be used by Scala 3 services. If your library on Scala 2 uses the `ducks` library and your Scala 3 application uses `ducks` as well, you get two conflicting versions `ducks_2.13` and `ducks_3`.
+
+To solve this, with sbt, you can cross-publish your library to both versions by adding one line to your build:
+
+```scala
++ crossScalaVersions := List("2.13.12", "3.3.1")
+```
+
+To build against all versions listed in `crossScalaVersions`, prefix the action to run with `+`. For example:
+
+```
+sbt> + compile
+```
+
+*At least, that’s what we did — everything else just worked. Your mileage may vary.*
+
+> For more information, see the guides on [ClassPath compatibility](https://docs.scala-lang.org/scala3/guides/migration/compatibility-classpath.html) and [Cross-building (sbt)](https://www.scala-sbt.org/1.x/docs/Cross-Build.html) or [Cross Builds (mill)](https://mill-build.com/mill/0.9.12/Cross_Builds.html)
+
+### Dealing with dependencies
+
+If all the libraries you use are actively maintained and published for Scala 3, good, there is nothing for you to worry about.
+
+Otherwise, you might need [to help them](https://docs.scala-lang.org/contribute/) with migration or eliminate the dependency. Some libraries can be redundant on Scala 3 (because, for instance, Scala 3 has support for [Type Class Derivation](https://docs.scala-lang.org/scala3/reference/contextual/derivation.html)), and some have new alternatives.
+
+*For example, at this stage, we’ve discovered [Iron](https://github.com/Iltotore/iron), [a great lightweight library](https://www.youtube.com/watch?v=4pq1elOap9k&ab_channel=ImpurePics) for type constraints, which replaced refined and newtype libraries in one go.*
+
+### Dealing with plugins
+
+Similar to dependencies, some plugins stay working, and some — are redundant and can be dropped.
+
+For instance, the first plugin I was happy to retire was `better-monadic-for`, because Scala 3 already gives me better for-comprehensions. For example, we can define implicit values inside for-comprehensions:
```

Comment:
"For instance" then "For example" is a duplicate. One of them should be dropped.